src/pybind/mgr/telemetry/module.py

```diff
@@ -1073,6 +1075,62 @@ def self_test(self) -> None:
         if 'report_id' not in report:
             raise RuntimeError('report_id not found in report')
 
+        #--------- Tests for the perf channel ---------------#
+
+        # Create a test pool. In command line, this is the same as
+        # `ceph osd pool create test_pool 100`
+        r, outb, outs = self.mon_command({
+                'prefix': 'osd pool create',
+                'pool': 'test_pool',
+                'pg_num': 100
+        })
+
+        # Get a copy of the perf report before any IO changes have been made:
+        perf_report_1 = self.compile_report(channels=['perf'])
+
+        # Set up a test file that will be uploaded to the test pool.
+        f = open('test_file.txt', 'w')
+        f.write("Hello World!")
+        f.close()
+
+        # Upload test_file to the test pool. In command line, this is the same as
+        # `rados -p test_pool put test_file test_file.txt`. Then, remove the text file.
+        subprocess.run(['rados', '-p', 'test_pool', 'put', 'test_file', 'test_file.txt'])
+        os.remove('test_file.txt')
+
+        # Give the rados command some time to execute.
+        time.sleep(5)
+
+        # Get a second report and compare it to the first report to see if any
+        # changes, particularly writes to test_pool, have been made.
+        perf_report_2 = self.compile_report(channels=['perf'])
+        # [TODO: INSERT COMPARISON HERE]
```

Comment:
I left a **TODO** here and in one other place because I haven't quite decided on how I want to do this part. But I thought it was a good idea to check that the completed parts are passing the tests.